File: frontend/src/app/sites/seven-wonders/dto/result/DuelResultTeamState.ts
import {Team} from "../../../../main/core/dto/Team";

export interface DuelResultTeamState {
  team: Team;
  score: number;
  blueCardScore: number;
  greenCardScore: number;
  yellowCardScore: number;
  purpleCardScore: number;
  wonderScore: number;
  scienceScore: number;
  coinScore: number;
  warScore: number;
  wonWithWar: boolean;
  wonWithScience: boolean;
}

File: backend/src/main/java/net/zorphy/backend/site/sevenwonders/dto/result/DuelResultTeamState.java
package net.zorphy.backend.site.sevenwonders.dto.result;

import net.zorphy.backend.site.core.http.dto.TeamDetails;
import net.zorphy.backend.site.core.http.dto.result.ResultStateTeamBase;

public record DuelResultTeamState(
        TeamDetails team,
        int score,
        int blueCardScore,
        int greenCardScore,
        int yellowCardScore,
        int purpleCardScore,
        int wonderScore,
        int scienceScore,
        int coinScore,
        int warScore,
        boolean wonWithWar,
        boolean wonWithScience
) implements ResultStateTeamBase {
}
